integration-test/src/test/java/org/ehcache/integration/ThreadPoolsTest.java

```diff
@@ -102,4 +104,23 @@ public void testDefaultThreadPoolFailsWithExplicitConfig() throws Exception {
     cacheManager.close();
   }
 
+  @Test
+  public void testNoDefaultThreadPoolFailsWithNoCacheEventDispatcher() throws Exception {
+    PooledExecutionServiceConfiguration executionServiceConfiguration = new PooledExecutionServiceConfiguration();
+    executionServiceConfiguration.addPool("foo", 2, 4);
+    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+      .using(executionServiceConfiguration)
+      .build(true);
+    try {
+      cacheManager.createCache("testCache",
+        CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class, heap(10))
+          .build());
+      fail("expected IllegalStateException");
+    } catch (IllegalStateException ise) {
+      String message = "Exception occurred in creating cache event dispatcher with error message : Null pool alias provided and no default pool configured";
+      assertEquals(message, ise.getCause().getMessage());
```

Comment:
Try to use Hamcrest `assertThat(...)` and the appropriate matchers when possible, it gives much better failure messages.